core/src/apps/common/storage/recovery.py

```diff
@@ -76,6 +79,33 @@ def get_slip39_iteration_exponent() -> Optional[int]:
     return common._get_uint8(_NAMESPACE, _SLIP39_ITERATION_EXPONENT)
 
 
+def set_slip39_group_count(group_count: int) -> None:
+    common._set_uint8(_NAMESPACE, _SLIP39_GROUP_COUNT, group_count)
+
+
+def get_slip39_group_count() -> Optional[int]:
+    return common._get_uint8(_NAMESPACE, _SLIP39_GROUP_COUNT)
+
+
+def set_slip39_group_threshold(group_threshold: int) -> None:
+    common._set_uint8(_NAMESPACE, _SLIP39_GROUP_THRESHOLD, group_threshold)
+
+
+def get_slip39_group_threshold() -> Optional[int]:
+    return common._get_uint8(_NAMESPACE, _SLIP39_GROUP_THRESHOLD)
+
+
+# TODO: better names?
+def set_slip39_shares_remaining(shares_remaining: bytearray) -> None:
+    common._set(_NAMESPACE, _SLIP39_SHARES_REMAINING, shares_remaining)
+
+
+def get_slip39_shares_remaining() -> Optional[bytearray]:
+    """ returns bytearray of remaining shares. result[0] = remaining shares for group 1
+        default value for shares that haven't been entered yet is 0x10 """
+    return common._get(_NAMESPACE, _SLIP39_SHARES_REMAINING)
```

Comment:
0 is a valid value that we reach when the number of shares reaches the threshold of that particular group. So it would be ambiguous to use it as default value that is set if we don't have any shares from that group. I decided to use the max possible threshold (16/0x10) as the default. 

I will rework it to return a List though, that will probably be less confusing. If we decide that we don't care about recovery storage compatibility for upgrade on beta we can drop the original geters/seters for simple shamir too, it would simplify the rest of the code too. .

Reply:
Oh that makes sense. Just please move it to a constant and add a comment.

Reply:
Done. Based on IRL discussion we also decided to reuse _REMAINING storage key. It used to be uint8 now we're storing a lot more there. Could this cause possible issues when upgrading firmware during recovery?